src/clients/FtpGateway.cc

```diff
@@ -2579,7 +2579,7 @@ Ftp::Gateway::haveParsedReplyHeaders()
     // makePublic() if allowed/possible or release() otherwise
     if (flags.authenticated || // authenticated requests can't be cached
             getCurrentOffset() ||
-            !e->makePublic()) {
+            !e->makePublic(false)) {
         e->release();
     }
```

Comment:
We have discussed that added `makePublic(false)` calls deserve analysis. I think this particular case matches the suggested default failure handling -- the caller does want to share this otherwise-cachable entry among existing readers, even if it cannot be made `ksDefault` public. That also implies that this caller does not want to call `release(void)` if `makePublic()` fails because that call would make the entry not just private but non-sharable.

On the other hand, the `release()` call is needed for the other two `if` statement conditions and, hence, cannot be simply removed.

I suggest the following refactoring:

```suggestion
if (flags.authenticated || getCurrentOffset()) {
    e->release();
    return;
}
e->makePublic();
```

I do not think we need existing C++ comments -- this code is self-documenting enough IMO.